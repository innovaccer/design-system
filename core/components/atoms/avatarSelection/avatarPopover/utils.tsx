import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
