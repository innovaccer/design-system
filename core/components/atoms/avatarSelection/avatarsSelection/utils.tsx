import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
