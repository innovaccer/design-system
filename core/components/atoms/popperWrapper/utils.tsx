// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
