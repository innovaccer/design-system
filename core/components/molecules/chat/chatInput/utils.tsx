import { MessageType } from './ChatInput';

export const positionMentionPopup = (
  textareaRef: React.RefObject<HTMLDivElement>,
  setMentionPosition: React.Dispatch<React.SetStateAction<{ top: number; left: number }>>
) => {
  if (!textareaRef.current) return;

  const selection = window.getSelection();
  if (!selection || !selection.rangeCount) return;

  const range = selection.getRangeAt(0);
  const rect = range.getBoundingClientRect();
  const editorRect = textareaRef.current.getBoundingClientRect();

  // Calculate position based on the cursor position
  let top = rect.bottom + window.scrollY;
  let left = rect.left + window.scrollX;

  // Adjust position if it would go off-screen
  if (top + 150 > window.innerHeight) {
    top = rect.top + window.scrollY - 150;
  }
  if (left + 200 > window.innerWidth) {
    left = window.innerWidth - 200;
  }

  // Adjust position relative to the editor
  top -= editorRect.top + window.scrollY - 12;
  left -= editorRect.left + window.scrollX - 8;

  setMentionPosition({ top, left });
};

export const extractMessageData = (textareaRef: React.RefObject<HTMLDivElement>): MessageType[] => {
  const elements = textareaRef.current?.childNodes;

  return Array.from(elements || []).map((item: any) => {
    if (item?.dataset?.type === 'mention') {
      const mentionValue = JSON.parse(item.dataset.content);
      return { type: 'mention', content: mentionValue?.data, nodeName: item.nodeName };
    }

    // handles new line by pressing enter
    else if (item.innerText === '\n' && item.nodeName === 'DIV') {
      return { type: 'newline', content: '\n', nodeName: item.nodeName };
    }

    // handles line break by pressing shift + enter
    else if (item.nodeName === 'BR') {
      return { type: 'linebreak', content: '<br>', nodeName: item.nodeName };
    }

    return { type: 'text', content: item.innerText || item.textContent, nodeName: item.nodeName };
  });
};
