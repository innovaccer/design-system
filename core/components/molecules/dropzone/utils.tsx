const isIe = (userAgent: string) => {
  return userAgent.indexOf('MSIE') !== -1 || userAgent.indexOf('Trident/') !== -1;
};

const isEdge = (userAgent: string) => {
  return userAgent.indexOf('Edge/') !== -1;
};

export const accepts = (file: File, acceptedFiles?: string | string[]) => {
  if (file && acceptedFiles) {
    const acceptedFilesArray = Array.isArray(acceptedFiles) ? acceptedFiles : acceptedFiles.split(',');
    const fileName = file.name || '';
    const mimeType = (file.type || '').toLowerCase();
    const baseMimeType = mimeType.replace(/\/.*$/, '');

    if (!mimeType) {
      return true;
    }

    return acceptedFilesArray.some((type) => {
      const validType = type.trim().toLowerCase();
      if (validType.charAt(0) === '.') {
        return fileName.toLowerCase().endsWith(validType);
      }
      if (validType.endsWith('/*')) {
        return baseMimeType === validType.replace(/\/.*$/, '');
      }
      return mimeType === validType;
    });
  }
  return true;
};

export const isPropagationStopped = (event: any) => {
  if (typeof event.isPropagationStopped === 'function') {
    return event.isPropagationStopped();
  }
  if (typeof event.cancelBubble !== 'undefined') {
    return event.cancelBubble;
  }
  return false;
};

export const isEvtWithFiles = (event: any) => {
  if (!event.dataTransfer) {
    return !!event.target && !!event.target.files;
  }

  return Array.prototype.some.call(
    event.dataTransfer.types,
    (type) => type === 'Files' || type === 'application/x-moz-file'
  );
};

export const isKindFile = (item: null | Record<string, unknown>) => {
  return typeof item === 'object' && item !== null && item.kind === 'file';
};

export const onDocumentDragOver = (event: DragEvent) => {
  event.preventDefault();
};

export const isIeOrEdge = (userAgent = window.navigator.userAgent) => {
  return isIe(userAgent) || isEdge(userAgent);
};

type functype = (event: Event, ...args: unknown[]) => void;

export const composeEventHandlers = (...fns: functype[]) => {
  return (event: Event, ...args: unknown[]) =>
    fns.some((fn) => {
      if (!isPropagationStopped(event) && fn) {
        fn(event, ...args);
      }
      return isPropagationStopped(event);
    });
};

export const reducer = (state: any, action: any) => {
  switch (action.type) {
    case 'focus':
      return {
        ...state,
        isFocused: true,
      };
    case 'blur':
      return {
        ...state,
        isFocused: false,
      };
    case 'openDialog':
      return {
        ...state,
        isFileDialogActive: true,
      };
    case 'closeDialog':
      return {
        ...state,
        isFileDialogActive: false,
      };
    case 'setDraggedFiles':
      /* eslint no-case-declarations: 0 */
      const { isDragActive, draggedFiles } = action;
      return {
        ...state,
        draggedFiles,
        isDragActive,
      };
    case 'setFiles':
      return {
        ...state,
        acceptedFiles: action.acceptedFiles,
        fileRejections: action.fileRejections,
      };
    case 'reset':
      return {
        ...state,
        isFileDialogActive: false,
        isDragActive: false,
        draggedFiles: [],
        acceptedFiles: [],
        fileRejections: [],
      };
    default:
      return state;
  }
};
