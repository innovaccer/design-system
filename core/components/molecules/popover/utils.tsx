import { Placement } from '@floating-ui/react';
import { PositionType as Position } from '@/common.type';

export const mapPositionToPlacement = (position: Position): Placement => {
  const positionMap: Record<Position, Placement> = {
    auto: 'bottom',
    'auto-start': 'bottom-start',
    'auto-end': 'bottom-end',
    'top-start': 'top-start',
    top: 'top',
    'top-end': 'top-end',
    'right-start': 'right-start',
    right: 'right',
    'right-end': 'right-end',
    'bottom-end': 'bottom-end',
    bottom: 'bottom',
    'bottom-start': 'bottom-start',
    'left-end': 'left-end',
    left: 'left',
    'left-start': 'left-start',
  };

  return positionMap[position] || 'bottom';
};
