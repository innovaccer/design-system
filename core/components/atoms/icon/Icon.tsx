import * as React from 'react';
import classNames from 'classnames';
import { BaseProps, extractBaseProps } from '@/utils/types';

export type Appearance = 'default' | 'destructive' | 'white' | 'subtle' | 'disabled' | 'info' | 'alert' | 'warning' | 'success';
export type IconType = 'filled'
  | 'outlined'
  | 'outline'
  | 'rounded'
  | 'round'
  | 'two-tone'
  | 'sharp';      // 'outline', 'rounded' to be deprecated soon.

export interface IconProps extends BaseProps {
  /**
   * Material icon name
   */
  name: string;
  /**
   * Size of `Icon`
   */
  size: number;
  /**
   * Type of material `Icon`
   *
   * ** `'outline' | 'rounded'` will be deprecated**
   */
  type?: IconType;
  /**
   * Color of `Icon`
   */
  appearance?: Appearance;
  /**
   * Handler to be called when icon is clicked
   */
  onClick?: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void;
}

export const Icon = (props: IconProps) => {
  const {
    appearance,
    className,
    name,
    size,
    onClick
  } = props;

  const baseProps = extractBaseProps(props);

  const mapper = (val: IconProps['type']) => {
    if (val === 'outline') return 'outlined';
    if (val === 'rounded') return 'round';
    return val;
  };

  const type = mapper(props.type);

  const iconClass = classNames({
    ['material-icons']: true,       // change to !type || type === 'filled' after migration
    [`material-icons-${mapper(type)}`]: type && type !== 'filled',
    ['Icon']: true,
    [`Icon--${appearance}`]: appearance,
    [`${className}`]: className
  });

  const styles = {
    fontSize: `${size}px`,
    width: `${size}px`,
  };

  // change `children` to {name} after migration
  return (
    <i
      {...baseProps}
      className={iconClass}
      style={styles}
      onClick={onClick}
    >
      {type ? `${name}_${type}` : name}
    </i>
  );
};

Icon.displayName = 'Icon';
Icon.defaultProps = {
  size: 16
};

export default Icon;
