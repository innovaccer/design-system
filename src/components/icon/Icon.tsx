import * as React from 'react';
import classNames from 'classnames';
import { BaseProps, extractBaseProps } from '@/utils/types';
import { useAccessibilityProps } from '@/accessibility/utils';

export type IconAppearance =
  | 'default'
  | 'destructive'
  | 'white'
  | 'subtle'
  | 'disabled'
  | 'alert'
  | 'warning'
  | 'success'
  | 'primary_lighter'
  | 'primary'
  | 'primary_dark'
  | 'alert_lighter'
  | 'alert_dark'
  | 'warning_lighter'
  | 'warning_dark'
  | 'success_lighter'
  | 'success_dark'
  | 'accent1'
  | 'accent1_lighter'
  | 'accent1_dark'
  | 'accent2'
  | 'accent2_lighter'
  | 'accent2_dark'
  | 'accent3'
  | 'accent3_lighter'
  | 'accent3_dark'
  | 'accent4'
  | 'accent4_lighter'
  | 'accent4_dark'
  | 'inverse';
export type IconType = 'filled' | 'outlined' | 'rounded' | 'two-tone' | 'sharp';

export interface IconProps extends BaseProps {
  /**
   * Size of `Icon`
   */
  size: number;
  /**
   * Type of material `Icon`
   *
   */
  type?: IconType;
  /**
   * Color of `Icon`
   */
  appearance?: IconAppearance;
  /**
   * Handler to be called when icon is clicked
   */
  onClick?: (e: React.MouseEvent<HTMLElement>) => void;
  /**
   * Handler to be called when key is pressed on icon
   */
  onKeyDown?: (e: React.KeyboardEvent<HTMLElement>) => void;
  /**
   * DOM node to be passed as child to the component
   */
  children?: React.ReactNode;
  /**
   * The tabindex global attribute indicates that its element can be focused, and
   * where it participates in sequential keyboard navigation.
   */
  tabIndex?: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>['tabIndex'];
}

export const Icon = (props: IconProps) => {
  const { appearance, className, size, children, type, tabIndex } = props;
  const accessibilityProps = useAccessibilityProps(props);

  const baseProps = extractBaseProps(props);

  const getIconAppearance = (iconColor: string) => {
    const x = iconColor.indexOf('_');
    return iconColor.slice(0, x) + iconColor.charAt(x + 1).toUpperCase() + iconColor.slice(x + 2);
  };

  const color = appearance && appearance.includes('_') ? getIconAppearance(appearance) : appearance;

  const iconClass = classNames(
    {
      ['material-icons']: true, // change to !type || type === 'filled' after migration
      [`material-icons-${type}`]: type && type !== 'filled',
      ['Mds-Icon']: true,
      [`Mds-Icon--${color}`]: appearance,
    },
    className
  );

  const styles = {
    fontSize: `${size}px`,
    width: `${size}px`,
  };

  return (
    <i {...baseProps} className={iconClass} style={styles} {...accessibilityProps} tabIndex={tabIndex}>
      {type ? `${children}_${type}` : children}
    </i>
  );
};

Icon.displayName = 'Icon';
Icon.defaultProps = {
  size: 16,
  type: 'round',
};

export default Icon;
