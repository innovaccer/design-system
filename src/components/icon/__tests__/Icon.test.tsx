import * as React from 'react';
import { render } from '@testing-library/react';
import Icon, { IconProps as Props } from '../Icon';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const size = 50;
const appearance = [
  'destructive',
  'white',
  'subtle',
  'disabled',
  'alert',
  'success',
  'warning',
  'primary',
  'primary_dark',
  'primary_lighter',
  'alert_dark',
  'alert_lighter',
  'success_dark',
  'success_lighter',
  'warning_dark',
  'warning_lighter',
  'accent1',
  'accent1_dark',
  'accent1_lighter',
  'accent2',
  'accent2_dark',
  'accent2_lighter',
  'accent3',
  'accent3_dark',
  'accent3_lighter',
  'accent4',
  'accent4_dark',
  'accent4_lighter',
  'inverse',
];
const type = ['filled', 'outlined', 'round', 'two-tone', 'sharp'];
const FunctionValue = jest.fn();
const StringValue = 'events';

describe('Icon component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(StringValue, { required: true }),
    size: valueHelper(size, { required: true }),
    appearance: valueHelper(appearance, { required: true, iterate: true }),
    onClick: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Icon {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };
  testHelper(mapper, testFunc);
});

describe('Icon component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(StringValue, { required: true }),
    size: valueHelper(size, { required: true }),
    type: valueHelper(type, { required: true, iterate: true }),
    onClick: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Icon {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };
  testHelper(mapper, testFunc);

  describe('Unit tests', () => {
    it('add material icon type class when type is present and is filled', () => {
      const { getByTestId } = render(
        <Icon data-test="DesignSystem--Icon" type="outlined" size={50}>
          info
        </Icon>
      );
      expect(getByTestId('DesignSystem--Icon').className).toContain('material-icons-outlined');
    });

    it('to contain material-icon-round class when type is not present', () => {
      const { getByTestId } = render(
        <Icon data-test="DesignSystem--Icon" size={50}>
          info
        </Icon>
      );
      expect(getByTestId('DesignSystem--Icon').className).toContain('material-icons-round');
    });

    it('add icon appearance class when appearance is present', () => {
      const { getByTestId } = render(
        <Icon data-test="DesignSystem--Icon" size={50} appearance="accent1">
          info
        </Icon>
      );
      expect(getByTestId('DesignSystem--Icon').className).not.toContain('Mds-Icon-accent1');
    });
  });
});
