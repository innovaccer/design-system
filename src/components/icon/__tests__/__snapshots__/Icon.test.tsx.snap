// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component 
  name: "events", size: 50, appearance: "accent1", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent1"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent1_dark", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent1Dark"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent1_lighter", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent1Lighter"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent2", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent2"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent2_dark", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent2Dark"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent2_lighter", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent2Lighter"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent3", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent3"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent3_dark", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent3Dark"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent3_lighter", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent3Lighter"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent4", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent4"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent4_dark", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent4Dark"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "accent4_lighter", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--accent4Lighter"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "alert", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--alert"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "alert_dark", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--alertDark"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "alert_lighter", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--alertLighter"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "destructive", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--destructive"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "disabled", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--disabled"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "inverse", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--inverse"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "primary", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--primary"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "primary_dark", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--primaryDark"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "primary_lighter", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--primaryLighter"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "subtle", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--subtle"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "success", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--success"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "success_dark", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--successDark"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "success_lighter", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--successLighter"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "warning", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--warning"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "warning_dark", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--warningDark"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "warning_lighter", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--warningLighter"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, appearance: "white", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--white"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, type: "filled", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons Mds-Icon"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_filled
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, type: "outlined", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-outlined Mds-Icon"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_outlined
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, type: "round", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-round Mds-Icon"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_round
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, type: "sharp", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-sharp Mds-Icon"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_sharp
    </i>
  </div>
</body>
`;

exports[`Icon component 
  name: "events", size: 50, type: "two-tone", onClick: "[Function]"
 1`] = `
<body>
  <div>
    <i
      class="material-icons material-icons-two-tone Mds-Icon"
      role="button"
      style="font-size: 50px; width: 50px;"
    >
      undefined_two-tone
    </i>
  </div>
</body>
`;
